src/OpenTK.Windowing.GraphicsLibraryFramework/Input/JoystickState.cs

```diff
@@ -197,42 +201,27 @@ public override string ToString()
 
         internal unsafe void Update()
         {
-            UpdateHats();
-
-            UpdateAxes();
-
-            UpdateButtons();
-        }
-
-        private unsafe void UpdateButtons()
-        {
-            Utils.Swap(ref _buttons, ref _buttonsPrevious);
-
-            var b = GLFW.GetJoystickButtonsRaw(Id, out int count);
-            for (var j = 0; j < count; j++)
+            var hats = GLFW.GetJoystickHats(Id);
```

Comment:
Nitpick: Assuming this returns a span, can we cast that span instead of doing it for each item in the loop?